File: frontend/src/models/pregnancy.ts
/**
 * TypeScript models for pregnancy tracking
 */

export type PregnancyStatus = 'pregnant' | 'delivered' | 'miscarried'

export interface Pregnancy {
  id: string
  mother_id: string
  father_id: string
  conceived_at: string
  due_at: string
  status: PregnancyStatus
  progress_percentage: number // 0-100
  time_remaining_seconds: number
  is_due: boolean
  created_at?: string
  updated_at?: string
}

export interface PregnancyProgress {
  id: string
  mother_id: string
  father_id: string
  status: PregnancyStatus
  conceived_at: string
  due_at: string
  progress_percentage: number
  time_remaining_seconds: number
  is_due: boolean
}

export interface DeliveryResult {
  pregnancy_id: string
  child_id: string
  message: string
}

File: backend/app/schemas/pregnancy.py
"""Pydantic schemas for pregnancy management."""

from datetime import datetime

from pydantic import UUID4, ConfigDict
from sqlmodel import SQLModel

from app.models.pregnancy import PregnancyBase
from app.schemas.common import PregnancyStatusEnum


class PregnancyCreate(SQLModel):
    """Schema for creating a new pregnancy."""

    mother_id: UUID4
    father_id: UUID4


class PregnancyUpdate(SQLModel):
    """Schema for updating a pregnancy."""

    status: PregnancyStatusEnum | None = None
    due_at: datetime | None = None


class PregnancyRead(PregnancyBase):
    """Schema for reading a pregnancy."""

    id: UUID4
    progress_percentage: float
    time_remaining_seconds: int
    is_due: bool

    model_config = ConfigDict(from_attributes=True, use_enum_values=True)


class PregnancyProgress(SQLModel):
    """Pregnancy progress information."""

    id: UUID4
    mother_id: UUID4
    father_id: UUID4
    status: PregnancyStatusEnum
    conceived_at: datetime
    due_at: datetime
    progress_percentage: float
    time_remaining_seconds: int
    is_due: bool

    model_config = ConfigDict(use_enum_values=True)


class DeliveryResult(SQLModel):
    """Result of a pregnancy delivery."""

    pregnancy_id: UUID4
    child_id: UUID4
    message: str
